src/ISLogger.cpp

```diff
@@ -129,23 +129,41 @@ void cISLogger::Cleanup()
 
 void cISLogger::Update()
 {
-    // if we have a timeout param and the time has elapsed, shutdown
+    time_t timeSec = GetTime();
+
     if (m_lastCommTime == 0)
     {
-        m_lastCommTime = GetTime();
+        m_lastCommTime = timeSec;
     }
-    else if (m_timeoutFlushSeconds > 0 && GetTime() - m_lastCommTime >= m_timeoutFlushSeconds)
+    else if (m_timeoutFlushSeconds > 0 && timeSec - m_lastCommTime >= m_timeoutFlushSeconds)
     {
         for (auto it : m_devices)
         {
             it.second->Flush();
         }
     }
+
+    if (m_enabled && m_maxDiskSpace!=0 && 
+        m_timeoutFileCullingSeconds > 0 && 
+        timeSec - m_lastFileCullingTime >= m_timeoutFileCullingSeconds)
+    {   // File culling
+        m_lastFileCullingTime = timeSec;
+
+        // Update amount of drive space for parent log directory
+        m_usedDiskSpace = ISFileManager::GetDirectorySpaceUsed(m_rootDirectory);
+
+        if (m_usedDiskSpace > m_maxDiskSpace)
+        {   // Need to remove files
+           ISFileManager::RemoveOldestFiles(m_rootDirectory, m_maxDiskSpace);
+           ISFileManager::RemoveEmptyDirectories(m_rootDirectory);
+        }
+    }
+
     ISFileManager::TouchFile(m_directory + "/stats.txt");
 }
 
 
-bool cISLogger::InitSaveCommon(eLogType logType, const string &directory, const string &subDirectory, float maxDiskSpacePercent, uint32_t maxFileSize, bool useSubFolderTimestamp)
+bool cISLogger::InitSaveCommon(eLogType logType, const string &directory, const string &subDirectory, float driveUsageLimitPercent, float driveUsageLimitMb, uint32_t maxFileSize, bool useSubFolderTimestamp)
```

Comment:
This breaks the API.  Its probably fine, but if possible we should add new arguments to the end, with defaults, so that exist code still compiles and runs.  Alternatively, we should probably get away from these types of calls with lots of parameters, and move to "ISLogger::SetDiveUsageLimitPercent(float pct);" and the like, with a final "ISLogger::InitSave( ... )" with only the non-optional arguments passed.

Reply:
The latest version introduces new a InitSave() function while keeping the old versions and labeling them as deprecated.